src/ResourceManager/Reservations/Commands.Reservations/Cmdlets/PatchReservation.cs

```diff
@@ -63,5 +73,68 @@ public override void ExecuteCmdlet()
                 WriteObject(response);
             }
         }
+
+        private void PreRegister(string subscriptionId)
+        {
+            try
+            {
+                IAzureContext context;
+                if (TryGetDefaultContext(out context)
+                    && context.Account != null
+                    && context.Subscription != null)
+                {
+                    var client = new ResourceManagementClient(
+                                context.Environment.GetEndpointAsUri(AzureEnvironment.Endpoint.ResourceManager),
+                                AzureSession.Instance.AuthenticationFactory.GetServiceClientCredentials(context, AzureEnvironment.Endpoint.ResourceManager));
+                    client.SubscriptionId = subscriptionId;
```

Comment:
Here I am setting client.SubscriptionId as one that is passed as argument to this cmdlet; It is not from default context. This is because we want to be able to pre-register for any subscription that is under the same tenant.. it does not have to be default subscription or billing subscription. To my understanding setting client.SubscriptionId here will cause client.Providers.Get to get the providers for that subscription. Please let me know if I am mistaken.